vms/platformvm/state/stakers.go

```diff
@@ -99,6 +99,10 @@ type baseStakers struct {
 	// subnetID --> nodeID --> current state for the validator of the subnet
 	validators map[ids.ID]map[ids.NodeID]*baseStaker
 	stakers    *btree.BTreeG[*Staker]
+
+	// ordered view of each subnet validator set
+	subnetValidators map[ids.ID]*btree.BTreeG[*Staker]
```

Comment:
I'd add a `GetStakerBySubnetIterator` method in `CurrentStakers` only and use `subnetValidators` to implement it. Seems to me it may be helpful in other place in the codebase, in lieu of `GetCurrentStakerIterator`

Reply:
In general this all effort of your @joshua-kim, seems to indicate that we should try and support more general iterations over the stakers set.